File: java/parser/src/main/java/gov/lbl/parser/domain/Assignment_with_function_call_statement.java
package gov.lbl.parser.domain;

public class Assignment_with_function_call_statement {
    private Output_expression_list output_expression_list;
    private Component_reference function_name;
    private Function_call_args function_call_args;

    public Assignment_with_function_call_statement(Output_expression_list output_expression_list, Component_reference function_name, Function_call_args function_call_args) {
        this.output_expression_list = output_expression_list;
        this.function_name = function_name;
        this.function_call_args = function_call_args;
    }
}

File: json2mo/assignmentWithFunctionCallStatement.js
function parse (content, rawJson = false) {
  const componentReferenceParser = require('./componentReference')
  const outputExpressionListParser = require('./outputExpressionList')
  const functionCallArgsParser = require('./functionCallArgs')

  let moOutput = ''
  moOutput += '('
  if (content.output_expression_list != null) {
    moOutput += outputExpressionListParser.parse(content.output_expression_list, rawJson)
  }
  moOutput += ')'
  moOutput += ':='
  if (content.function_name != null) {
    moOutput += componentReferenceParser.parse(content.function_name, rawJson)
  }
  if (content.function_call_args != null) {
    moOutput += functionCallArgsParser.parse(content.function_call_args, rawJson)
  }
  return moOutput
}

module.exports = { parse }
